src/wallet/rpcwallet.cpp

```diff
@@ -1064,6 +1065,153 @@ UniValue sendmany(const JSONRPCRequest& request)
+        const UniValue& optval = request.params[0][optname];
+        if (optname == "privkeys") {
+            const UniValue& privkeys_a = optval.get_array();
+            for (size_t privkey_i = 0; privkey_i < privkeys_a.size(); ++privkey_i) {
+                const UniValue& privkey_wif = privkeys_a[privkey_i];
+                std::string wif_secret = privkey_wif.get_str();
+                CKey key;
+                CPubKey pubkey;
+                ParseWIFPrivKey(wif_secret, key, &pubkey);
+
+                tempKeystore.AddKey(key);
+                CKeyID address = pubkey.GetID();
+                CScript script = GetScriptForDestination(address);
+                if (!script.empty()) {
+                    needles.insert(script);
+                }
+                script = GetScriptForRawPubKey(pubkey);
+                if (!script.empty()) {
+                    needles.insert(script);
+                }
+            }
+        } else if (optname == "label") {
+            label = AccountFromValue(optval.get_str());
+        } else {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Unrecognised option '%s'", optname));
+        }
+    }
+
+    // Ensure keypool is filled if possible
+    {
+        LOCK2(cs_main, pwallet->cs_wallet);
+
+        if (!pwallet->IsLocked()) {
+            pwallet->TopUpKeyPool();
+        }
+    }
+
+    // Reserve the key we will be using
+    CReserveKey reservekey(pwallet);
+    CPubKey pubkey;
+    if (!reservekey.GetReservedKey(pubkey)) {
+        throw JSONRPCError(RPC_WALLET_KEYPOOL_RAN_OUT, "Error: Keypool ran out, please call keypoolrefill first");
+    }
+
+    // Scan UTXO set for inputs
+    std::vector<CTxOut> input_txos;
+    {
+        // Collect all possible inputs
+        std::map<COutPoint, Coin> coins;
+        {
+            LOCK(cs_main);
+            mempool.FindScriptPubKey(needles, coins);
+            FlushStateToDisk();
+            pcoinsdbview->FindScriptPubKey(needles, coins);
+        }
+
+        // Add them as inputs to the transaction, and count the total value
+        for (auto& it : coins) {
+            const COutPoint& outpoint = it.first;
+            const Coin& coin = it.second;
+            const CTxOut& txo = coin.out;
+            tx.vin.emplace_back(outpoint.hash, outpoint.n);
+            input_txos.push_back(txo);
+            total_in += txo.nValue;
+        }
+    }
+
+    if (total_in == 0) {
+        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "No value to sweep");
+    }
+
+    CKeyID keyID = pubkey.GetID();
+    CTxDestination txdest = CBitcoinAddress(keyID).Get();
+
+    tx.vout.emplace_back(total_in, GetScriptForDestination(txdest));
+
+    while (true) {
+        if (IsDust(tx.vout[0], ::minRelayTxFee)) {
+            throw JSONRPCError(RPC_VERIFY_REJECTED, "Swept value would be dust");
+        }
+        for (size_t input_index = 0; input_index < tx.vin.size(); ++input_index) {
```

Comment:
Fee changes require resigning. Perhaps it could be made to sign only before and after the loop, but the time here is trivial in comparison to the UTXO search.